src/idl_gen_grpc.cpp

```diff
@@ -165,53 +171,260 @@ class FlatBufFile : public grpc_generator::File {
 
   std::unique_ptr<const grpc_generator::Service> service(int i) const {
     return std::unique_ptr<const grpc_generator::Service> (
-          new FlatBufService(parser_.services_.vec[i]));
+             new FlatBufService(parser_.services_.vec[i]));
   }
 
   std::unique_ptr<grpc_generator::Printer> CreatePrinter(std::string *str) const {
     return std::unique_ptr<grpc_generator::Printer>(
-          new FlatBufPrinter(str));
+             new FlatBufPrinter(str));
   }
 
- private:
+private:
   const Parser &parser_;
   const std::string &file_name_;
 };
 
-bool GenerateGRPC(const Parser &parser,
-                  const std::string &/*path*/,
-                  const std::string &file_name) {
-
-  int nservices = 0;
-  for (auto it = parser.services_.vec.begin();
-       it != parser.services_.vec.end(); ++it) {
-    if (!(*it)->generated) nservices++;
-  }
-  if (!nservices) return true;
+bool GenerateCppGRPC(const Parser &parser,
+                     const std::string &/*path*/,
+                     const std::string &file_name) {
 
   grpc_cpp_generator::Parameters generator_parameters;
   // TODO(wvo): make the other parameters in this struct configurable.
   generator_parameters.use_system_headers = true;
-
-  FlatBufFile fbfile(parser, file_name);
+  grpc_cpp::FlatBufFile fbfile(parser, file_name);
 
   std::string header_code =
-      grpc_cpp_generator::GetHeaderPrologue(&fbfile, generator_parameters) +
-      grpc_cpp_generator::GetHeaderIncludes(&fbfile, generator_parameters) +
-      grpc_cpp_generator::GetHeaderServices(&fbfile, generator_parameters) +
-      grpc_cpp_generator::GetHeaderEpilogue(&fbfile, generator_parameters);
+    grpc_cpp_generator::GetHeaderPrologue(&fbfile, generator_parameters) +
+    grpc_cpp_generator::GetHeaderIncludes(&fbfile, generator_parameters) +
+    grpc_cpp_generator::GetHeaderServices(&fbfile, generator_parameters) +
+    grpc_cpp_generator::GetHeaderEpilogue(&fbfile, generator_parameters);
 
   std::string source_code =
-      grpc_cpp_generator::GetSourcePrologue(&fbfile, generator_parameters) +
-      grpc_cpp_generator::GetSourceIncludes(&fbfile, generator_parameters) +
-      grpc_cpp_generator::GetSourceServices(&fbfile, generator_parameters) +
-      grpc_cpp_generator::GetSourceEpilogue(&fbfile, generator_parameters);
+    grpc_cpp_generator::GetSourcePrologue(&fbfile, generator_parameters) +
+    grpc_cpp_generator::GetSourceIncludes(&fbfile, generator_parameters) +
+    grpc_cpp_generator::GetSourceServices(&fbfile, generator_parameters) +
+    grpc_cpp_generator::GetSourceEpilogue(&fbfile, generator_parameters);
 
   return flatbuffers::SaveFile((file_name + ".grpc.fb.h").c_str(),
                                header_code, false) &&
          flatbuffers::SaveFile((file_name + ".grpc.fb.cc").c_str(),
                                source_code, false);
 }
 
+}// namespace grpc_cpp
+
+namespace grpc_go {
+
+class FlatBufMethod : public grpc_generator::Method {
```

Comment:
these classes appear identical to the ones used for C++, why are they duplicated?